src/main/docs/guide/config/i18n.adoc

```diff
@@ -0,0 +1,31 @@
+== Internationalization (i18n)
+
+Micronaut provides for internationalization by reading user defined properties files.  A resource bundle is the grouping
+of all properties files with the same base name.  All locales translations will provide the keys the same but the values
+will be in the language specified.
+
+messages.properties - file will define key value pairs like this:
+
+[source,text]
+app.title=micronaut.demo
+
+To retrieve this message inside your application, you will use the ResourceBundle like this:
+
+[source,java]
+package com.example;
+import io.micronaut.runtime.Micronaut;
+public class Application {
+    public static void main(String[] args) {
+        Micronaut.run(Application.class);
+        Locale defaultLocale = new Locale();
+        ResourceBundle messages = ResourceBundle.getBundle("messages", defaultLocale);
+        String msg1 = messages.getString("app.title");
+        System.out.println( msg1 ); // this will print "micronaut.demo"
+    }
+}
+
+Micronaut provides a class called
+link:{api}/io/micronaut/context/i18n/ResourceBundleMessageSource.html[ResourceBundleMessageSource] to read the user
```

Comment:
`link:{api}/io/micronaut/context/i18n/ResourceBundleMessageSource.html[ResourceBundleMessageSource]`can be simplified as `api:context.i18n.ResourceBundleMessageSource[]`.